sdk/core/azure-core-amqp/src/models/amqp_header.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#include "azure/core/amqp/models/amqp_header.hpp"
+#include "azure/core/amqp/models/amqp_value.hpp"
+
+#include <azure_uamqp_c/amqp_definitions_milliseconds.h>
+
+#include <azure_uamqp_c/amqp_definitions_header.h>
+#include <chrono>
+#include <iostream>
+
+void Azure::Core::_internal::UniqueHandleHelper<HEADER_INSTANCE_TAG>::FreeAmqpHeader(
+    HEADER_HANDLE handle)
+{
+  header_destroy(handle);
+}
+
+namespace Azure { namespace Core { namespace Amqp { namespace Models {
+  bool MessageHeader::operator==(MessageHeader const& that) const noexcept
+  {
+    return this->DeliveryCount == that.DeliveryCount && this->Durable == that.Durable
+        && this->IsFirstAcquirer == that.IsFirstAcquirer && this->Priority == that.Priority
+        && this->TimeToLive.HasValue() == that.TimeToLive.HasValue()
+        && (this->TimeToLive.HasValue() ? this->TimeToLive.Value() == that.TimeToLive.Value()
+                                        : true);
+  }
+
+  MessageHeader _internal::MessageHeaderFactory::FromUamqp(UniqueMessageHeaderHandle const& handle)
+  {
+    MessageHeader rv;
+    bool boolValue;
+    if (!header_get_durable(handle.get(), &boolValue))
+    {
+      rv.Durable = boolValue;
+    }
+
+    uint8_t uint8Value;
+    if (!header_get_priority(handle.get(), &uint8Value))
+    {
+      rv.Priority = uint8Value;
+    }
+
+    milliseconds millisecondsValue;
+    if (!header_get_ttl(handle.get(), &millisecondsValue))
+    {
+      rv.TimeToLive = std::chrono::milliseconds(millisecondsValue);
+    }
+
+    if (!header_get_first_acquirer(handle.get(), &boolValue))
+    {
+      rv.IsFirstAcquirer = boolValue;
+    }
+
+    uint32_t uint32Value;
+    if (!header_get_delivery_count(handle.get(), &uint32Value))
+    {
+      rv.DeliveryCount = uint32Value;
+    }
+    return rv;
+  }
+
+  UniqueMessageHeaderHandle _internal::MessageHeaderFactory::ToUamqp(MessageHeader const& header)
+  {
+    UniqueMessageHeaderHandle rv{header_create()};
+    if (header.Durable)
+    {
+      if (header_set_durable(rv.get(), header.Durable))
+      {
+        throw std::runtime_error("Could not set durable value.");
+      }
```

Comment:
When could that happen?

Reply:
The exception will be thrown if the uAMQP `header_set_durable` API fails.

I don't know why it might fail, and honestly, it doesn't matter why it might fail - the API call is defined as possibly failing, and since it's defined as possibly fail, I need to code defensively in case it fails.